.github/workflows/mypy.yaml

```diff
@@ -0,0 +1,51 @@
+name: "mypy"
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    branches:
+      - develop
+
+concurrency:
+
+  group: "${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}"
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  mypy :
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: setup_miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: 3.8
+          auto-update-conda: true
+          channel-priority: flexible
+          channels: conda-forge
+          add-pip-as-python-dependency: true
+          mamba-version: "*"
+          architecture: x64
+
+      - name: install_deps
+        uses: ./.github/actions/setup-deps
+        with:
+          mamba: true
+          full-deps: true
```

Comment:
Given that the 2.3.0 release will likely support a minimum numpy version of 1.20 and that type annotations were added to numpy in 1.20, it would be better to set this to 1.20.0.

I'll submit a PR today or tomorrow raising our minimum numpy requirements to 1.20 for everything else.

Reply:
ok mentor.